### PR TITLE
cmd/snap: add --terminate flag to remove command

### PR DESCRIPTION
**Pull request by ZeyadYasser**
This PR exposes the ability to kill running snap apps through `snap remove --terminate`.

This PR is part 4 of implementing force removal of snaps according to [SD174 - Force removal of snaps](https://docs.google.com/document/d/1ikDI-vPJO6YYFHOyFJ5hfK5L5Gv8voMeY8HJvxNWujw/edit#bookmark=id.3mwbudc7fx0j) spec.

This the last part of a series PRs to implement killing running snap apps during removal, and cannot be merged before all previous PRs are merged (marking as blocked for now):
- Part 1: https://github.com/snapcore/snapd/pull/14126
- Part 2: https://github.com/snapcore/snapd/pull/14144
- Part 3.1: https://github.com/snapcore/snapd/pull/14160
- ~~Part 3.2: https://github.com/snapcore/snapd/pull/14189~~
- Part 3.2: https://github.com/canonical/snapd/pull/14354

### CODE REVIEW

File: cmd/snap/cmd_snap_op.go [hunk 1]
```diff
@@ -1396,6 +1397,8 @@ func init() {
 			"revision": i18n.G("Remove only the given revision"),
 			// TRANSLATORS: This should not start with a lowercase letter.
 			"purge": i18n.G("Remove the snap without saving a snapshot of its data"),
+			// TRANSLATORS: This should not start with a lowercase letter.
+			"terminate": i18n.G("Kill running snap apps"),
```
Comment (zyga) [inline cmd/snap/cmd_snap_op.go:1401]: Perhaps something less voilent: 
```suggestion
			"terminate": i18n.G("Terminate running application processes"),
```

**Review by zyga — APPROVED** [review]
LGTM with a suggestion on wording.

**Review by andrewphelpsj — APPROVED** [review]
Thanks!

**Comment by ernestl** [comment]
This was marked blocked because merging this will enable feature that is not yet complete. 
Once https://github.com/snapcore/snapd/pull/14189 is merged, this may be merged if spread coverage is deemed sufficient (may want to add additional test).

File: cmd/snap/cmd_snap_op.go (hunk 1 above)
Comment (ZeyadYasser) [reply cmd/snap/cmd_snap_op.go]: Updated, Thanks!

**Comment by ZeyadYasser** [comment]
Dependent PR is merged, and this PR can now be merged. Waiting for test run.

**Comment by ZeyadYasser** [comment]
Only `openstack:fedora-40-64:tests/main/selinux-clean` is failing, which is known to fail and is tracked.

**Review by degville — APPROVED** [review]
Looks great, thank you! Just one small typo and a comment.

**Comment by ZeyadYasser** [comment]
skipping spread tests for small typo fix https://github.com/canonical/snapd/pull/14192#discussion_r1757020650

File: cmd/snap/cmd_snap_op.go [hunk 2]
```diff
@@ -1493,6 +1494,8 @@ func init() {
 			"revision": i18n.G("Remove only the given revision"),
 			// TRANSLATORS: This should not start with a lowercase letter.
 			"purge": i18n.G("Remove the snap without saving a snapshot of its data"),
+			// TRANSLATORS: This should not start with a lowercase letter.
+			"terminate": i18n.G("Terminate running application processe"),
```
[the remark this answers is not in the document]
Comment (ZeyadYasser) [reply cmd/snap/cmd_snap_op.go]: Thanks for catching this! Updated

**Pull request merged**